zookeeper/source/zk_nn_client.cc

```diff
@@ -1548,21 +1546,35 @@ void ZkNnClient::get_block_locations(const std::string &src,
       LOG(INFO) << "[get_block_locations] Breaking at block " << sorted_block;
       break;
     }
-    if (size + block_size >= offset) {
-      auto data = std::vector<uint8_t>();
-      if (!zk->get(zk_path + "/" + sorted_block, data,
-                   error_code, sizeof(uint64_t))) {
-        LOG(ERROR) << "[get_block_locations] Failed to get "
-                   << zk_path << "/"
-                   << sorted_block
-                   << " info: "
-                   << error_code;
-        return;  // TODO(2016): Signal error
-      }
-      uint64_t block_id = *reinterpret_cast<uint64_t *>(&data[0]);
-      LOG(INFO) << "[get_block_locations] Found block " << block_id
-                << " for " << zk_path;
 
+    // Retreive block size
```

Comment:
If we have time later, probably should modularize this process out (very low priority for now).

Reply:
I think a refactor now will be good. This may actually be related to a bug that we saw (cc @eddiedugan @adamnsm1)

Reply:
This function might not work for EC, but we can take care of modifying it for that. Also I'm beginning to think we need to do a refactor for get_block_metadata_path so it can tell for ec whether we're asking for a block group or a storage block